tests-functional/tests/test_wakuext_group_chats.py

```diff
@@ -0,0 +1,192 @@
+from uuid import uuid4
+import pytest
+from steps.messenger import MessengerSteps
+from resources.enums import MessageContentType
+
+
+@pytest.mark.rpc
+class TestCreatePrivateGroups(MessengerSteps):
+
+    @pytest.fixture(autouse=True)
+    def setup_backends(self, backend_factory):
+        """Initialize two unprivileged backends (sender and receiver) for each test function"""
+        self.sender = backend_factory("sender")
+        self.receiver = backend_factory("receiver")
+        self.make_contacts(self.sender, self.receiver)
+
+    def test_create_group_chat_with_members(self):
+        private_group_name = f"private_group_{uuid4()}"
+        create_group_response = self.sender.wakuext_service.create_group_chat_with_members([self.receiver.public_key], private_group_name)
+        self.sender.verify_json_schema(create_group_response, method="wakuext_createGroupChatWithMembers")
+
+        self.get_message_by_content_type(
+            create_group_response,
+            content_type=MessageContentType.SYSTEM_MESSAGE_CONTENT_PRIVATE_GROUP.value,
+            message_pattern=f"@{self.sender.public_key} created the group {private_group_name}",
+        )[0]
+        self.get_message_by_content_type(
+            create_group_response,
+            content_type=MessageContentType.SYSTEM_MESSAGE_CONTENT_PRIVATE_GROUP.value,
+            message_pattern=f"@{self.sender.public_key} has added @{self.receiver.public_key}",
+        )[0]
+
+    def test_leave_group_chat(self):
+        create_group_response = self.sender.wakuext_service.create_group_chat_with_members([self.receiver.public_key], f"private_group_{uuid4()}")
+
+        group_id = create_group_response.get("result", {}).get("chats", [])[0].get("id")
+        leave_group_response = self.sender.wakuext_service.leave_group_chat(group_id, True)
+        self.sender.verify_json_schema(leave_group_response, method="wakuext_leaveGroupChat")
+
+        self.get_message_by_content_type(
```

Comment:
I think it's not enough just to check the system message, but also check that the group is gone from the chat list.